lib/socket.js

```diff
@@ -263,6 +264,11 @@ internals.Socket.prototype._lifecycle = function (request) {
         throw Boom.badRequest('Message missing id');
     }
 
+    if (!this._verifyAuth()) {
```

Comment:
I only added this for received messages, because of the heartbeat. If the server receives no heartbeat - it will disconnect anyways. There is a chance of someone configuring a very long heartbeat timeout and keeping on pushing messages, but is that an issue?